src/helpers/daily-data-providers/combined-sleep.ts

```diff
@@ -0,0 +1,42 @@
+import { add } from "date-fns";
+import { appleHealthSleepDataProvider, fitbitTotalSleepMinutesDataProvider, garminTotalSleepMinutesDataProvider } from ".";
+import getDayKey from "../get-day-key";
+import MyDataHelps from "@careevolution/mydatahelps-js";
+
+export default function (startDate: Date, endDate: Date) {
+    let providers: Promise<{ [key: string]: number }>[] = [];
+
+    return MyDataHelps.getDataCollectionSettings().then((settings) => {
+        if (settings.fitbitEnabled) {
+            providers.push(fitbitTotalSleepMinutesDataProvider(startDate, endDate));
+        }
+        if (settings.garminEnabled) {
+            providers.push(garminTotalSleepMinutesDataProvider(startDate, endDate));
+        }
+        if (settings.queryableDeviceDataTypes.find(s => s.namespace == "AppleHealth" && s.type == "SleepAnalysisInterval")) {
+            providers.push(appleHealthSleepDataProvider(startDate, endDate));
+        }
+
+        if (!providers.length) {
+            return {};
+        }
+
+        return Promise.all(providers).then((values) => {
+            var data: { [key: string]: number } = {};
+            while (startDate < endDate) {
+                var dayKey = getDayKey(startDate);
+                var sleep: number | null = null;
+                values.forEach((value) => {
+                    if (value[dayKey] || sleep == null || sleep < value[dayKey]) {
```

Comment:
Should this be: `value[dayKey] && (sleep == null || sleep < value[dayKey])`?

Reply:
huh...yeah, that would certainly make more sense.  That's interesting, it's the same across steps / RHR combined providers as well, so likely we have been displaying lower values than we should be for those combined providers (steps would be the major one).

Reply:
Well, just for steps I guess, RHR uses average